doc/qbk/3.1.url.qbk

```diff
@@ -46,49 +52,66 @@ sections that follow.
 
 [snippet_accessing_1]
 
+[heading URL types]
+
 The library uses the convention that each function `parse_<component>`
 operates according to the particular grammar rule `<component>`
 specified in __rfc3986__. Thus, it provides `parse_uri` for the
 [@https://datatracker.ietf.org/doc/html/rfc3986#section-3 ['URI]]
-grammar rule defined in __rfc3986__. A URL string can be parsed
-using one of the following parsing functions.
+grammar rule defined in __rfc3986__. A URL string can also be parsed
+using one of the following functions:
 
 [table Parsing Functions [
     [Function]
     [Grammar]
+    [ABNF]
     [Example]
     [Notes]
 ][
     [[link url.ref.boost__urls__parse_uri `parse_uri`]]
     [[@https://datatracker.ietf.org/doc/html/rfc3986#section-3 ['URI]]]
+    [[teletype]`scheme ":" hier-part [ "?" query ] [ "#" fragment ]`]
     [[teletype]`http://www.boost.org/index.html?field=value#downloads`]
     [Supports fragment `#downloads`]
 ][
     [[link url.ref.boost__urls__parse_absolute_uri `parse_absolute_uri`]]
     [[@https://datatracker.ietf.org/doc/html/rfc3986#section-4.3 ['absolute-URI]]]
+    [[teletype]`scheme ":" hier-part [ "?" query ]`]
     [[teletype]`http://www.boost.org/index.html?field=value`]
     [Does not support fragment]
-][
-    [[link url.ref.boost__urls__parse_origin_form `parse_origin_form`]]
-    [[@https://datatracker.ietf.org/doc/html/rfc7230#section-5.3.1 ['origin-form]]]
-    [[teletype]`/index.html?field=value`]
-    [Used in HTTP requests]
 ][
     [[link url.ref.boost__urls__parse_relative_ref `parse_relative_ref`]]
     [[@https://datatracker.ietf.org/doc/html/rfc3986#section-4.2 ['relative-ref]]]
+    [[teletype]`relative-part [ "?" query ] [ "#" fragment ]`]
     [[teletype]`//www.boost.org/index.html?field=value#downloads`]
     [Does not require scheme or authority]
 ][
     [[link url.ref.boost__urls__parse_uri_reference `parse_uri_reference`]]
     [[@https://datatracker.ietf.org/doc/html/rfc3986#section-4.1 ['URI-reference]]]
+    [[teletype]`URI / relative-ref`]
     [[teletype]`http://www.boost.org/index.html`]
     [Any `URI` or `relative-ref`]
+][
+    [[link url.ref.boost__urls__parse_origin_form `parse_origin_form`]]
```

Comment:
but now it is not in alphabetical order?

Reply:
Like the RFC, I tried to sort them by their "didactic importance".

- Absolute URIs are the most common type of URL. The semantics of all other types refer to absolute URLs.
- It's only possible to understand the semantics relative_ref after absolute URLs.
- uri_reference is just one of the two above.
- parse_origin_form is not even the same spec.

Reply:
hmm... ok